src/main_app/jobs_workers/admin_jobs_workers/collect_templates_data/worker.py

```
@@ -226,26 +235,44 @@ def _process_one(self, template: TemplateRecord) -> bool:
+                template_data["last_world_file"] = last_file
             else:
                 template_info.steps.last_world_file._update(result="skipped", msg="No changes")
 
         # ------------------
-        # template_info step # 3 source
+        # template_info step # 2 newest_year
+        try:
+            newest_year = find_newest_year(wikitext)
+            if not newest_year:
+                raise Exception("Could not find newest year")
+        except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)